### PR TITLE
x3270: Make X11 dependency optional

### PR DESCRIPTION
**Pull request by ronmrdechai**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/x3270.rb [hunk 1]
```diff
@@ -10,17 +10,18 @@ class X3270 < Formula
...
   end
 
+  option "without-x11", "Exclude x3270 (X11-based version)"
   option "with-c3270", "Include c3270 (curses-based version)"
   option "with-s3270", "Include s3270 (displayless version)"
   option "with-tcl3270", "Include tcl3270 (integrated with Tcl)"
   option "with-pr3287", "Include pr3287 (printer emulation)"
 
-  depends_on :x11
+  depends_on :x11 => :recommended
```
Comment (MikeMcQuaid) [inline Formula/x3270.rb]: This should be `:optional`

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/x3270.rb [hunk 2]
```diff
@@ -10,17 +10,18 @@ class X3270 < Formula
     sha256 "897d18de2b85172e751719687710f20b1717293b10dac669d6eb052e08d4b02a" => :yosemite
   end
 
+  option "without-x11", "Exclude x3270 (X11-based version)"
```
Comment (MikeMcQuaid) [inline Formula/x3270.rb]: Probably wants flipped to `with-x11` now, sorry, should have noted that last time.

Comment (ronmrdechai) [inline Formula/x3270.rb]: If do that a build without arguments will fail (`brew install x3270`), an argument must be supplied. Is that okay?

Comment (MikeMcQuaid) [inline Formula/x3270.rb]: > Also, the name should probably be with-x3270 then, and if so, how do I bind a dependency to an option?

You can't, unfortunately. This is a limitation of our dependency system.

Go for:
`option "with-x11", "Include x3270 (X11-based version)"`

**Comment by MikeMcQuaid** [comment]
Install seems to be failing now: https://bot.brew.sh/job/Homebrew%20Core%20Pull%20Requests/427/version=sierra/testReport/junit/brew-test-bot/sierra/install_x3270/

**Comment by ronmrdechai** [comment]
Yes that's the issue with making X11 opt-in. The configure script has nothing to build (actually the issue is with a missing X11 dependency, but never mind that for now), so I'm not really sure what it's doing.
Making one of the build options opt-out and the others opt-in fixes this. This is more of an issue with the way this package's configure script works than anything.
Enabling X11 by default seems like the right thing to do as the package's name is x3270 and the X-based binary is named `x3270`, other binaries are named `c3270`, `s3280`, etc. What do you think?

**Comment by MikeMcQuaid** [comment]
Ah, I see now what you meant earlier, sorry! I think enabling everything that doesn't require a dependency/requirement (i.e. everything except X11) would be good if possible otherwise picking either the console or ncurses one would be good. Thoughts?

**Comment by ronmrdechai** [comment]
Seems right to me.

**Comment by MikeMcQuaid** [comment]
Will need to adjust the `test do` accordingly for a default binary. Also, squash commits by running `brew update; git checkout $YOUR_BRANCH; git rebase --interactive origin/master` and then mark each commit other than the first as `squash` or `fixup` then `git push -f`. Thanks!

**Comment by ronmrdechai** [comment]
Done, thank you.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks again for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**